wgpu-types/src/lib.rs

```diff
@@ -5463,8 +5463,16 @@ pub struct SurfaceConfiguration<V> {
     /// `Bgra8Unorm` and `Bgra8UnormSrgb`
     pub format: TextureFormat,
     /// Width of the swap chain. Must be the same size as the surface, and nonzero.
+    ///
+    /// If this is not the same size as the underlying surface (e.g. if it is
+    /// set once, and the window is later resized), the behaviour is defined
+    /// but platform-specific, and may change in the future.
```

Comment:
I'm actually leaning towards going even further and start listing out the status quo per platform. Drawback of that is ofc that it crosslinks the docs with a piece of independent code and may thus easily outdate, also we'd be starting with just MacOS which just looks a bit bad.
It is not too unusual for us though to point out behavior of specific platforms in here since the benefits (user not having to dig deeply) outweigh the cost (maintenance, risk of outdated information). Usually we do so indirectly by referring to features and abilities, but that's not possible here obviously.

This is an improvement regardless, so I'd leave it to your judgement how much we want to leak the implementation details here.

Reply:
I felt like it was a lot of detail to put into a fairly niche area (not resizing when the documentation says you should do) - it gives it undue "weight" compared to the rest of the documentation, almost as-if we support such use-cases where I really don't want to - but that may be just a feeling, a small note like the following would probably be fine:

```suggestion
    /// set once, and the window is later resized), the behaviour is defined
    /// but platform-specific, and may change in the future (currently macOS
    /// scales the surface, other platforms may do something else).
```

Reply:
Not a strong stance, but I would prefer the small note you propose :) 👍